cpp/src/aztec/honk/composer/standard_honk_composer.test.cpp

```diff
@@ -0,0 +1,86 @@
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+    composer.create_add_gate({ a_idx, b_idx, c_idx, fr::one(), fr::one(), fr::neg_one(), fr::zero() });
+
+    auto proving_key = composer.compute_proving_key();
+    barretenberg::fr left = barretenberg::fr::one();
+    barretenberg::fr right = barretenberg::fr::one();
+    // Let's check that indices are the same and nothing is lost, first
+    for (size_t i = 0; i < composer.program_width; i++) {
+
+        std::string index = std::to_string(i + 1);
+        auto polynomial = proving_key->polynomial_cache.get("sigma_" + index + "_lagrange");
+        for (size_t j = 0; j < proving_key->n; j++) {
+            left *= i * (proving_key->n) + j;
+            right *= polynomial.coefficients[j];
+        }
+    }
+
+    EXPECT_EQ(left, right);
+    barretenberg::fr beta = barretenberg::fr::random_element();
+    barretenberg::fr gamma = barretenberg::fr::random_element();
+    left = barretenberg::fr::one();
+    right = barretenberg::fr::one();
+    // Now let's check that witness values correspond to the permutation
+
+    composer.compute_witness();
+    for (size_t i = 0; i < composer.program_width; i++) {
+
+        std::string index = std::to_string(i + 1);
+        auto permutation_polynomial = proving_key->polynomial_cache.get("sigma_" + index + "_lagrange");
+        auto witness_polynomial = proving_key->polynomial_cache.get("w_" + index + "_lagrange");
+        for (size_t j = 0; j < proving_key->n; j++) {
+            auto current_witness = witness_polynomial.coefficients[j];
```

Comment:
This is cool.

Reply:
Checking that the permutation works?

Reply:
The quick-and-dirty product argument in a test. Meta!